easybuild/easyblocks/l/lammps.py

```diff
@@ -224,46 +346,87 @@ def configure_step(self, **kwargs):
-            self.cfg.update('configopts', '-DKOKKOS_ARCH="%s"' % get_kokkos_arch(cuda_cc, self.cfg['kokkos_arch']))
+                self.cfg.update('configopts', '-D%s_ENABLE_OPENMP=yes' % self.kokkos_prefix)
 
             # if KOKKOS and CUDA
-            if cuda:
+            if self.cuda:
                 nvcc_wrapper_path = os.path.join(self.start_dir, "lib", "kokkos", "bin", "nvcc_wrapper")
-                self.cfg.update('configopts', '-DKOKKOS_ENABLE_CUDA=yes')
-                self.cfg.update('configopts', '-DCMAKE_CXX_COMPILER="%s"' % nvcc_wrapper_path)
-                self.cfg.update('configopts', '-DCMAKE_CXX_FLAGS="-ccbin $CXX $CXXFLAGS"')
+                self.cfg.update('configopts', '-D%s_ENABLE_CUDA=yes' % self.kokkos_prefix)
+                if LooseVersion(self.cur_version) >= LooseVersion(self.ref_version):
+                    self.cfg.update('configopts', '-D%s_ARCH_%s=yes' % (self.kokkos_prefix, processor_arch))
+                    self.cfg.update('configopts', '-D%s_ARCH_%s=yes' % (self.kokkos_prefix, gpu_arch))
+                else:
+                    # Older versions of Kokkos required us to tweak the C++ compiler
+                    self.cfg.update('configopts', '-DCMAKE_CXX_COMPILER="%s"' % nvcc_wrapper_path)
+                    self.cfg.update('configopts', '-DCMAKE_CXX_FLAGS="-ccbin $CXX $CXXFLAGS"')
+                    self.cfg.update('configopts', '-D%s_ARCH="%s;%s"' % (self.kokkos_prefix, processor_arch, gpu_arch))
+            else:
+                if LooseVersion(self.cur_version) >= LooseVersion(self.ref_version):
+                    self.cfg.update('configopts', '-D%s_ARCH_%s=yes' % (self.kokkos_prefix, processor_arch))
+                else:
+                    self.cfg.update('configopts', '-D%s_ARCH="%s"' % (self.kokkos_prefix, processor_arch))
 
         # CUDA only
-        elif cuda:
-            self.cfg.update('configopts', '-D%sGPU=on' % PKG_PREFIX)
+        elif self.cuda:
+            self.cfg.update('configopts', '-D%sGPU=on' % self.pkg_prefix)
             self.cfg.update('configopts', '-DGPU_API=cuda')
             self.cfg.update('configopts', '-DGPU_ARCH=%s' % get_cuda_gpu_arch(cuda_cc))
 
+        # Make sure that all libraries end up in the same folder (python libs seem to default to lib, everything else
+        # to lib64)
+        self.cfg.update('configopts', '-DCMAKE_INSTALL_LIBDIR=lib')
+
         # avoid that pip (ab)uses $HOME/.cache/pip
         # cfr. https://pip.pypa.io/en/stable/reference/pip_install/#caching
         env.setvar('XDG_CACHE_HOME', tempfile.gettempdir())
         self.log.info("Using %s as pip cache directory", os.environ['XDG_CACHE_HOME'])
 
+        # Make sure it uses the Python we want
+        python_dir = get_software_root('Python')
+        if python_dir:
+            python_short_version = '.'.join(get_software_version('Python').split('.')[:2])
+            python_lib = glob.glob("%s/lib*/libpython%s.so" % (python_dir, python_short_version))[0]
```

Comment:
For Python 3.7 (2019b) the library is `libpython3.7m.so`m, which is why I see https://github.com/easybuilders/easybuild-easyblocks/pull/2213#issuecomment-1192264481

Reply:
Goes off to recall how to include an optional m

Reply:
```
python -c "import sysconfig; print(sysconfig.get_config_var('abiflags'))"
```

Is from https://github.com/easybuilders/easybuild-easyblocks/blob/develop/easybuild/easyblocks/p/python.py#L511 - i.e. how we check for the right library in the Python easyblock.

Reply:
Urgh, I should have waited before committing https://github.com/easybuilders/easybuild-easyblocks/pull/2213/commits/9f9591b5191b525ceb56afef077d1029df1a2d02, will fix now